File: 백준/Bronze/16466. 콘서트/콘서트.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.HashSet;
import java.util.Set;

public class Main {
    public static int findSmallestTicket(int N, int[] soldTickets) {
        Set<Integer> soldSet = new HashSet<>();
        for (int ticket : soldTickets) {
            soldSet.add(ticket);
        }
        
        int ticketNumber = 1;
        while (soldSet.contains(ticketNumber)) {
            ticketNumber++;
        }
        
        return ticketNumber;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(reader.readLine());
        int[] soldTickets = new int[N];
        String[] tokens = reader.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            soldTickets[i] = Integer.parseInt(tokens[i]);
        }
        reader.close();
        
        System.out.println(findSmallestTicket(N, soldTickets));
    }
}

File: 백준/Bronze/16466. 콘서트/콘서트.py
def find_smallest_ticket(N, sold_tickets):
    sold_set = set(sold_tickets)
    ticket_number = 1
    
    while ticket_number in sold_set:
        ticket_number += 1
    
    return ticket_number

# 입력 받기
import sys
input = sys.stdin.read
data = input().split()

N = int(data[0])
sold_tickets = list(map(int, data[1:]))

# 결과 출력
print(find_smallest_ticket(N, sold_tickets))
